src/linker/configuration.py

```diff
@@ -47,13 +53,66 @@ def get_implementation_config(self, step_name: str) -> Optional[Dict[str, Any]]:
     # Setup Methods #
     #################
 
+    def _get_schema(self) -> Optional[PipelineSchema]:
+        """Validates the pipeline against supported schemas."""
+
+        errors = defaultdict(dict)
+        for schema in PIPELINE_SCHEMAS:
+            logs = []
+            config_steps = self.pipeline["steps"].keys()
+            # Check that number of schema steps matches number of implementations
+            if len(schema.steps) != len(config_steps):
+                logs.append(
+                    f"Expected {len(schema.steps)} steps but found {len(config_steps)} implementations."
+                )
+            else:
+                for idx, config_step in enumerate(config_steps):
+                    # Check that all steps are accounted for and in the correct order
+                    schema_step = schema.steps[idx].name
+                    if config_step != schema_step:
+                        logs.append(
+                            f"Step {idx + 1}: the pipeline schema expects step '{schema_step}' "
+                            f"but the provided pipeline specifies '{config_step}'. "
+                            "Check step order and spelling in the pipeline configuration yaml."
+                        )
+            if logs:
+                errors["PIPELINE ERRORS"][schema.name] = logs
+                pass  # try the next schema
+            else:
+                return schema
+        # No schemas were validated
+        exit_with_validation_error(dict(errors))
```

Comment:
Why do we do this rather than just raising an exception?

Reply:
I wanted to hew to the error logging format that Steve had previously established.